cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -113,8 +119,218 @@ func (c *ZipDeployClient) BeginDeploy(
 	return runtime.NewPoller(response, c.pipeline, pollerOptions)
 }
 
+// Deploys the specified application zip to the azure app service using deployment status api and waits for completion
+func (c *ZipDeployClient) BeginDeployTrackStatus(
+	ctx context.Context,
+	zipFile io.Reader,
+	subscriptionId,
+	resourceGroup,
+	appName string,
+) (*runtime.Poller[armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse], error) {
+	request, err := c.createDeployRequest(ctx, zipFile)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := c.pipeline.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+
+	if !runtime.HasStatusCode(response, http.StatusAccepted) {
+		return nil, runtime.NewResponseError(response)
+	}
+
+	client, err := armappservice.NewWebAppsClient(subscriptionId, c.cred, &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Retry: policy.RetryOptions{
+				MaxRetries: -1},
+		},
+	})
```

Comment:
Here we're constructing the `arm.ClientOptions` from scratch. I think we should instead be inheriting the arm.ClientOptions default settings that is passed in during `NewZipDeployClient` instead of constructing one completely, I'll push a change shortly that introduces `c.armClientOptions`, but I'll leave it up to you here to wire it up.

Reply:
I pushed the change for `c.armClientOptions` that is ready for uptake.